server/requestlog/requestlog.go

```diff
@@ -25,7 +25,7 @@ import (
 	"net/http"
 	"time"
 
-	"go.opencensus.io/trace"
+	"go.opentelemetry.io/otel/trace"
```

Comment:
This doesn't look like it's using the bridge that IIUC is required to make this backwards compatible.

Reply:
https://pkg.go.dev/go.opentelemetry.io/otel/bridge/opentracing#section-readme

Reply:
@vangent Link you provide is talking about OpenTracing but SDK uses OpenCensus, take a look at this:
https://pkg.go.dev/go.opentelemetry.io/otel/bridge/opencensus
Anyway, this change will not be backwards compatible, because applications using the server package have to change the server `New` function parameter from OpenCensus to OpenTelemetry.

Reply:
I don't want to move ahead with this if it's not backwards compatible for users who still have OpenCensus monitoring integrations.

I'm a bit less concerned about the `New` call if it's straightforward for a user to update their code to compile and then it will continue to work with their OpenCensus integrations.

I'm more concerned about changing the application to no longer emit OpenCensus events, and only OpenTelemetry ones; that would require a user to update their monitoring stack as part of the upgrade. It's especially not OK to update some packages in this library in that way, but not others.

Reply:
The `New` call is straightforward enough for users IMO and I don't think users need to change their integrated monitoring facilities because OpenTelemetary is fully compatible in the case of exploring metrics and traces.

If bridge added, also all spans created with openTracing or OpenCensus will be exported by otel

Reply:
Also, known limitation to migrating is explained here:
https://opentelemetry.io/docs/migration/opentracing/#limits-on-compatibility

Reply:
Maybe I'm not understanding, but it seems like:

* Right now, with no changes, users can migrate to OpenTelemetry by creating a bridge as described in https://pkg.go.dev/go.opentelemetry.io/otel/bridge/opencensus ("Since OpenCensus telemetry uses globals, installing a bridge will result in telemetry collection from all libraries that use OpenCensus, including some you may not expect. For example ([#1928](https://github.com/open-telemetry/opentelemetry-go/issues/1928)), if a client library generates traces with OpenCensus, installing the bridge will cause those traces to be exported by OpenTelemetry."). They can use the bridge to create an OpenCensus to pass to `New` ("Override OpenCensus' DefaultTracer with the bridge").

So, with no changes, we currently support OpenCensus and OpenTelemetry (through a bridge).

* If we submit this CL now, then users who are using OpenTelemetry will have an easy switch by updating their call to `New`. However, users that are using OpenCensus will be forced to migrate; I don't think that OpenCensus dashboards will work with OpenTelemetry-only events.


Another way of saying this is looking at the migration journey:
https://opentelemetry.io/docs/migration/opentracing/#migration-overview

"Progressively re-instrument your application using the OpenTelemetry API. There is no hard requirement to remove existing OpenTracing API calls from your application, they will continue to work." is step 5 of 5; it happens after all of the other migration work is complete. I think this implies that shared libraries like this one should wait until a high percentage of its users have done steps 1-4.

Reply:
If we go by the OpenTelemetry and the bridge, users with OpenCensus tracing spans will still have access to their events.